logstash-core/lib/logstash/config/mixin.rb

```diff
@@ -99,6 +102,23 @@ def config_init(params)
       end
     end
 
+    # Resolve environment variables references
+    params.each do |name, value|
+      if (value.is_a?(Hash))
+        value.each do |valueHashKey, valueHashValue|
+          value[valueHashKey.to_s] = replace_env_placeholders(valueHashValue)
```

Comment:
It should be ok, in theory, if it's not, it probably just means we found a bug caused by something else within logstash going funny. :)